medicai/utils/inference.py

```
@@ -236,3 +236,284 @@ def sliding_window_inference(
+    padded_image_size = info["padded_image_size"]
+    original_image_size = info["original_image_size"]
+    pad_size = info["pad_size"]
+
+    output_image = None
+    count_map = None
+
+    for pred_batch, batch_slices, importance_map_resized in patch_generator:
+        # Lazy initialisation, infer num_classes from first prediction
+        if output_image is None:
+            nc = num_classes or pred_batch.shape[-1]
+            output_shape = [batch_size] + list(padded_image_size) + [nc]
```

Ruff (0.15.10)

[warning] Consider `[batch_size, *list(padded_image_size), nc]` instead of concatenation

Replace with `[batch_size, *list(padded_image_size), nc]`

(RUF005)